opnode/l2/api.go

```diff
@@ -0,0 +1,297 @@
+package l2
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/ethereum-optimism/optimistic-specs/opnode/eth"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/holiman/uint256"
+
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type ErrorCode int
+
+const (
+	UnavailablePayload ErrorCode = -32001
+)
+
+type Bytes32 [32]byte
+
+func (b *Bytes32) UnmarshalJSON(text []byte) error {
+	return hexutil.UnmarshalFixedJSON(reflect.TypeOf(b), text, b[:])
+}
+
+func (b *Bytes32) UnmarshalText(text []byte) error {
+	return hexutil.UnmarshalFixedText("Bytes32", text, b[:])
+}
+
+func (b Bytes32) MarshalText() ([]byte, error) {
+	return hexutil.Bytes(b[:]).MarshalText()
+}
+
+func (b Bytes32) String() string {
+	return hexutil.Encode(b[:])
+}
+
+type Bytes256 [256]byte
+
+func (b *Bytes256) UnmarshalJSON(text []byte) error {
+	return hexutil.UnmarshalFixedJSON(reflect.TypeOf(b), text, b[:])
+}
+
+func (b *Bytes256) UnmarshalText(text []byte) error {
+	return hexutil.UnmarshalFixedText("Bytes32", text, b[:])
+}
+
+func (b Bytes256) MarshalText() ([]byte, error) {
+	return hexutil.Bytes(b[:]).MarshalText()
+}
+
+func (b Bytes256) String() string {
+	return hexutil.Encode(b[:])
+}
+
+type Uint64Quantity = hexutil.Uint64
```

Comment:
Attempt to follow spec definitions with "quantity"/"data" type separation more closely. Since payload ID was uint64 but encoded as data instead of quantity. Once the upstream PR with payload-id type lands I can clean this up.